common/scripts/base.sh

```diff
@@ -89,23 +89,35 @@ function undeploy() {
   echoBold "Removing Marathon application: ${1}..."
   if ! dcos marathon app remove $1; then
     echoError "Failed to undeploy ${1}. Non-zero exit code returned from DCOS CLI"
+    return 1
   fi
   echoSuccess "Successfully undeployed ${1}"
   return 0
 }
 
-# Check whether given service port is open via marathon-lb
-# $1 - Marathon application ID
-# $2 - service port
+# Check whether given service port is open
+# $1 -Marathon application id
+# $2 -service port
 function waitUntilServiceIsActive() {
-  marathon_lb_host_ip=$(dcos marathon app show marathon-lb | $mesos_artifacts_home/common/scripts/get-marathon-lb-host.py)
-  echoBold "Waiting for ${1} to launch on ${marathon_lb_host_ip}:${2}..."
-  while ! nc -w3 $marathon_lb_host_ip $2 >/dev/null; do
-    echoBold "Waiting for ${1} to launch on ${marathon_lb_host_ip}:${2}..."
+  retry_count=12
+  count=0
+  while (! dcos marathon app show $1 | python $mesos_artifacts_home/common/scripts/get-host-ip.py $1 && [ "$count" -lt "$retry_count" ] ); do
+    echoBold "Waiting to get host ip for ${1}"
+    count=$((count + 1))
+    sleep 5s
```

Comment:
I think we can reduce the sleep period to 1s here since this is only to get the host ip